Sources/Tracing/Tracer.swift

```diff
@@ -315,12 +315,42 @@ public func withSpan<T>(
 ///   - instant: the time instant at which the span started
 ///   - context: The `ServiceContext` providing information on where to start the new ``Span``.
 ///   - kind: The ``SpanKind`` of the new ``Span``.
+///   - isolation: Defaulted parameter for inheriting isolation of calling actor
 ///   - function: The function name in which the span was started
 ///   - fileID: The `fileID` where the span was started.
 ///   - line: The file line where the span was started.
 ///   - operation: The operation that this span should be measuring
 /// - Returns: the value returned by `operation`
 /// - Throws: the error the `operation` has thrown (if any)
+#if swift(>=6.0)
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *) // for TaskLocal ServiceContext
+public func withSpan<T, Instant: TracerInstant>(
+    _ operationName: String,
+    at instant: @autoclosure () -> Instant,
+    context: @autoclosure () -> ServiceContext = .current ?? .topLevel,
+    ofKind kind: SpanKind = .internal,
+    isolation: isolated(any Actor)? = #isolation,
+    function: String = #function,
+    file fileID: String = #fileID,
+    line: UInt = #line,
+    _ operation: (any Span) async throws -> T
+) async rethrows -> T {
+    try await InstrumentationSystem.legacyTracer.withAnySpan(
+        operationName,
+        at: instant(),
+        context: context(),
+        ofKind: kind,
+        function: function,
+        file: fileID,
+        line: line
+    ) { anySpan in
+        try await operation(anySpan)
+    }
+}
+#endif
+
+@_disfavoredOverload
+@available(*, deprecated, message: "Prefer #isolation version of this API")
```

Comment:
Should the `deprecated` attribute be applied here when the `#isolation` API isn't available? Should this be in the same `#if` as above?